pkg/handler/storage/client.go

```diff
@@ -234,15 +242,39 @@ func checkRootSquash(nfs *openapi.NFSV2Spec) bool {
 	return true
 }
 
-func generateAttachmentList(in *openapi.StorageAttachmentV2Spec) []regionv1.Attachment {
-	out := make([]regionv1.Attachment, len(in.NetworkIDs))
-	for i := range in.NetworkIDs {
-		out[i] = regionv1.Attachment{
-			NetworkID: in.NetworkIDs[i],
+func generateAttachmentList(ctx context.Context, networkClient *network.Client, in *openapi.StorageAttachmentV2Spec) ([]regionv1.Attachment, error) {
+	networkIDs := in.NetworkIDs
+	out := make([]regionv1.Attachment, len(networkIDs))
+
+	for i, networkID := range networkIDs {
+		network, err := networkClient.GetV2Raw(ctx, networkID)
+		if err != nil {
+			return nil, errors.OAuth2ServerError("unable to get network").WithError(err)
+		}
+
+		attachment, err := generateAttachment(network)
+		if err != nil {
+			return nil, err
 		}
+
+		out[i] = *attachment
 	}
 
-	return out
+	return out, nil
+}
+
+func generateAttachment(network *regionv1.Network) (*regionv1.Attachment, error) {
+	networkID := network.Name
+
+	if network.Status.Openstack == nil {
```

Comment:
I think a comment on this would be helpful, especially since the comment on this type is "TODO: delete me"